packages/amplify-cli-core/src/state-manager/stateManager.ts

```diff
@@ -79,10 +79,11 @@ export class StateManager {
 
   getCustomPolicies = (categoryName: string, resourceName: string): CustomIAMPolicies | undefined => {
     const filePath = pathManager.getCustomPoliciesPath(categoryName, resourceName);
-    if (!(fs.existsSync(filePath)) || !isCustomPoliciesFile(filePath)) {
+    try{
+      return JSONUtilities.readJson<CustomIAMPolicies>(filePath);
+    } catch(err) {
```

Comment:
We need to handle empty file situation, if the file is empty, even throwIfNotExist equal to false, it will still throw an error ```
'jsonString' argument missing or empty
An error occurred during the push operation: 'jsonString' argument missing or empty
```
instead of returning undefined

Reply:
hmm okay. We should probably modify readJson to fix that, but this is fine for now